src/debugger/channel/AdapterChannel.js

```diff
@@ -51,7 +51,7 @@ export class AdapterChannel {
       this._eventEmitter.emit(DebugMessage.PREPACK_READY_RESPONSE, result);
       this.trySendNextRequest();
     } else if (messageType === DebugMessage.BREAKPOINT_ADD_ACKNOWLEDGE) {
-      let dbgResponse = this._marshaller.unmarshallBreakpointAddResponse(requestID);
+      let dbgResponse = this._marshaller.unmarshallBreakpointsAddResponse(requestID, parts.slice(2).join(" "));
```

Comment:
I can see this is violating the design of marshaller in several aspects:
1. All the parts related join slice parsing should move inside unmarshalXXX() call. This is true not only for this change, but all the other places you called slice() method.
2. I did not see why you need to parse parts and reqeustID fields from message at the beginning of "_processPrepackMessage". Please refactor this code to move all message parsing code into unmarshalXXX() API.
3. With above suggestions, unmarshalXXX() API should only take a single message as argument. So you only need a single marshal() public API which dispatchs its call to individual private _unmarshalXXX() APIs.